Support radial grids as cylindrical cells

magnesj opened the pull request:
Closes #12875

magnesj committed:
Add minimum angular count to system preferences

magnesj committed:
Refactors grid view updates and radial grid handling

Moves grid view update logic to RigReservoirGridTools for reuse and consistency.

Adds cylindrical coordinate import and radial grid refinement functionality.
This automatically creates a LGR for radial grids with low angular resolution, improving simulation accuracy.

magnesj committed:
Improves radial grid handling and LGR creation

Enhances the processing of radial grids by adding logic to create Local Grid Refinements (LGRs) when cylindrical coordinate conversion is enabled.

This involves checking for valid radial grid data, handling cases with invalid headers, and creating angular grid refinements based on user preferences.

Also ensures active cell indexing for multi-grid cases and updates views after grid modifications.

kriben commented on ApplicationLibCode/Application/RiaPreferencesSystem.cpp:
```diff
@@ -77,6 +77,11 @@ RiaPreferencesSystem::RiaPreferencesSystem()
     CAF_PDM_InitField( &m_showPdfExportDialog, "showPdfExportDialog", true, "Show PDF Export Dialog" );
     caf::PdmUiNativeCheckBoxEditor::configureFieldForEditor( &m_showPdfExportDialog );
 
+    CAF_PDM_InitField( &m_useCylindricalCoordinateConversion, "useCylindricalCoordinateConversion", true, "Use Cylindrical Coords" );
```
Use same name for member? Remove "conversion" part of name?

magnesj replied:
fixed

kriben commented on ApplicationLibCode/CommandFileInterface/RicfCreateLgrForCompletions.cpp:
```diff
@@ -121,7 +123,7 @@ caf::PdmScriptResponse RicfCreateLgrForCompletions::execute()
                                      &wellsIntersectingOtherLgrs );
 
-    feature->updateViews( eclipseCase );
+    RigReservoirGridTools::updateViews( eclipseCase );
```
Does more than just update view?

magnesj replied:
fixed

kriben commented on ApplicationLibCode/FileInterface/RifOpmRadialGridTools.cpp:
```diff
@@ -67,7 +73,74 @@ void RifOpmRadialGridTools::importCoordinatesForRadialGrid( const std::string& g
+
+        {
+            // Open the file and only check "GRIDHEAD" to be able to do an early return if no radial grids are present
+
+            Opm::EclIO::EclFile gridFile( gridFilePath );
+            auto                arrays = gridFile.getList();
+
+            int index = 0;
+            for ( const auto& [name, arrayType, arraySize] : arrays )
+            {
+                if ( name == "GRIDHEAD" )
+                {
+                    auto gridhead = gridFile.get<int>( index );
+                    if ( gridhead.size() > 26 && gridhead[26] > 0 )
+                    {
+                        isRadialGridPresent = true;
+                        break;
+                    }
+                }
+                index++;
+            }
+        }
+
```
lambda?

magnesj replied:
fixed

kriben reviewed (commented):
Some minor suggestions, but overall OK!

magnesj committed:
Cleanup based on review

magnesj committed:
Refactors radial grid import logic

Moves radial grid import logic into a separate function for improved code organization and readability.

This change encapsulates the radial grid import and refinement process, including handling of cylindrical coordinates and LGR creation, into a dedicated `checkAndImportRadialGrid` function. The original logic within `importGridAndResultMetaData` is replaced with a call to this new function, simplifying the import process.
